packages/@aws-cdk/aws-appmesh/test/integ.mesh.ts

```diff
@@ -99,22 +103,42 @@ const node3 = mesh.addVirtualNode('node3', {
 });
 
 router.addRoute('route-2', {
-  routeTargets: [
-    {
-      virtualNode: node2,
-      weight: 30,
+  routeSpec: appmesh.RouteSpec.http2({
+    weightedTargets: [
+      {
+        virtualNode: node2,
+        weight: 30,
+      },
+    ],
+    match: {
+      prefixPath: '/path2',
     },
-  ],
-  prefix: '/path2',
+  }),
 });
 
 router.addRoute('route-3', {
-  routeTargets: [
-    {
-      virtualNode: node3,
-      weight: 20,
+  routeSpec: appmesh.RouteSpec.tcp({
+    weightedTargets: [
+      {
+        virtualNode: node3,
+        weight: 20,
+      },
+    ],
+  }),
+});
+
+router.addRoute('route-4', {
+  routeSpec: appmesh.RouteSpec.grpc({
+    weightedTargets: [
+      {
+        virtualNode: node3,
+        weight: 20,
+      },
+    ],
+    match: {
+      serviceName: virtualService.virtualServiceName,
     },
-  ],
+  }),
```

Comment:
Let's keep this file as close to the original as possible (only make changes to make the code compile with the new APIs).